### PR TITLE
Support `Parsable` collections & scalar types in the request body

### PR DESCRIPTION
**Pull request by Ndiritu**
- support passing scalar values as the request body
- refactor setting collection request bodies

part of https://github.com/microsoft/kiota/issues/1933

### CODE REVIEW

File: src/RequestInformation.php [hunk 1]
```diff
@@ -147,25 +146,75 @@ public function setStreamContent(StreamInterface $value): void {
...
-    public function setContentFromParsable(RequestAdapter $requestAdapter, string $contentType, Parsable ...$values): void {
-        if (empty($values)) {
-            throw new InvalidArgumentException('$values cannot be empty.');
+    public function setContentFromParsable(RequestAdapter $requestAdapter, string $contentType, Parsable $value): void {
+        try {
+            $writer = $requestAdapter->getSerializationWriterFactory()->getSerializationWriter($contentType);
+            $writer->writeObjectValue(null, $value);
+            $this->headers[self::$contentTypeHeader] = $contentType;
+            $this->content = $writer->getSerializedContent();
+        } catch (Exception $ex) {
```
Comment (shemogumbe) [inline src/RequestInformation.php:161]: Could there be a way of making the variable name for the exception more meaningful in the context, and if not possible just something like exception//exception_message etc

File: src/RequestInformation.php [hunk 2]
```diff
@@ -147,25 +146,75 @@ public function setStreamContent(StreamInterface $value): void {
...
+     * @return void
+     */
+    public function setContentFromParsableCollection(RequestAdapter $requestAdapter, string $contentType, array $values): void
+    {
         try {
             $writer = $requestAdapter->getSerializationWriterFactory()->getSerializationWriter($contentType);
+            $writer->writeCollectionOfObjectValues(null, $values);
             $this->headers[self::$contentTypeHeader] = $contentType;
+            $this->content = $writer->getSerializedContent();
+        } catch (Exception $ex) {
```
Comment (shemogumbe) [inline src/RequestInformation.php:181]: Same as above

**Review by shemogumbe — APPROVED** [review]

File: src/RequestInformation.php [hunk 3]
```diff
@@ -147,25 +146,75 @@ public function setStreamContent(StreamInterface $value): void {
...
      * @param string $contentType the content type.
-     * @param Parsable ...$values the models.
+     * @param Parsable $value the models.
      */
-    public function setContentFromParsable(RequestAdapter $requestAdapter, string $contentType, Parsable ...$values): void {
-        if (empty($values)) {
-            throw new InvalidArgumentException('$values cannot be empty.');
+    public function setContentFromParsable(RequestAdapter $requestAdapter, string $contentType, Parsable $value): void {
+        try {
+            $writer = $requestAdapter->getSerializationWriterFactory()->getSerializationWriter($contentType);
```
Comment (MIchaelMainer) [inline src/RequestInformation.php:157]: What happens if getSerializationWriter() does not have a SerializationWriter for the contentType? I think that is a expected exception scenario. Consider adding an exception and/or message for that scenario so we can send a message that indicates we need a SerializationWriter to support the scenario. I think this comment is applicable to `setContentFromParsableCollection`, `setContentFromScalar`, and `setContentFromScalarCollection`.

Comment (MIchaelMainer) [reply src/RequestInformation.php:157]: I think I see the answer to my question here: 
https://github.com/microsoft/kiota-abstractions-php/blob/2af24f9b4e94b0e4f09201c3cdfb1cabe7e53cbc/src/Serialization/SerializationWriterFactoryRegistry.php#L46
https://github.com/microsoft/kiota-abstractions-php/blob/2af24f9b4e94b0e4f09201c3cdfb1cabe7e53cbc/src/Serialization/SerializationWriterFactoryRegistry.php#L52

What action should we expect a developer to do when this is encountered in a test environment? Consider making the error message more actionable.

`'Content type ' . $contentType . ' does not have a factory to be parsed'`

could be something like

`'Content type ' . $contentType . ' does not have a factory to be parsed. Add a SerializationWriter factory to support the content type.'`

Comment (Ndiritu) [reply src/RequestInformation.php:157]: Thanks for this. Updated the error message

**Pull request merged**